File: 3.x.x/crates-utopia/validators/src/wildcard.rs
use crate::{Validator, ValueType};
use serde_json::Value;

/// Always valid.
#[derive(Debug, Clone, Default)]
pub struct Wildcard;

impl Validator for Wildcard {
    fn description(&self) -> String {
        "Every input is valid".into()
    }

    fn value_type(&self) -> ValueType {
        ValueType::String
    }

    fn is_valid(&self, _value: &Value) -> bool {
        true
    }
}

File: packages/validators/src/Validator/Wildcard.php
<?php

declare(strict_types=1);

namespace Utopia\Validator;

use Utopia\Validator;

/**
 * Wildcard
 *
 * Does not perform any validation. Always returns valid
 */
class Wildcard extends Validator
{
    /**
     * Get Description
     *
     * Returns validator description
     */
    public function getDescription(): string
    {
        return 'Every input is valid';
    }

    /**
     * Is valid
     *
     * Validation will always pass irrespective of input
     *
     * @param  mixed  $value
     */
    public function isValid($value): bool
    {
        return true;
    }

    /**
     * Is array
     *
     * Function will return true if object is array.
     */
    public function isArray(): bool
    {
        return false;
    }

    /**
     * Get Type
     *
     * Returns validator type.
     */
    public function getType(): string
    {
        return self::TYPE_STRING;
    }
}
